app/controllers/application_controller/timelines.rb

```diff
@@ -105,42 +105,54 @@ def tl_build_timeline_report_options
       end
 
       temp_clause = @tl_record.event_where_clause(@tl_options.evt_type)
+      conditions = [temp_clause[0], "timestamp >= ?", "timestamp <= ?"]
+      parameters = temp_clause[1..-1] + [from_dt, to_dt]
 
-      cond = "( "
-      cond = cond << temp_clause[0]
-      params = temp_clause.slice(1, temp_clause.length)
-
-      event_set = @tl_options.event_set
-      if !event_set.empty?
-        if @tl_options.policy_events? && @tl_options.policy.result != "both"
-          where_clause = [") and (timestamp >= ? and timestamp <= ?) and (event_type in (?)) and (result = ?)",
-                          from_dt,
-                          to_dt,
-                          event_set.flatten,
-                          @tl_options.policy.result]
-        else
-          where_clause = [") and (timestamp >= ? and timestamp <= ?) and (event_type in (?))",
-                          from_dt,
-                          to_dt,
-                          event_set.flatten]
-        end
-      else
-        where_clause = [") and (timestamp >= ? and timestamp <= ?)",
-                        from_dt,
-                        to_dt]
-      end
-      cond << where_clause[0]
+      tl_add_event_type_conditions(conditions, parameters)
+      tl_add_policy_conditions(conditions, parameters) if @tl_options.policy_events?
 
-      params2 = where_clause.slice(1, where_clause.length - 1)
-      params = params.concat(params2)
-      @report.where_clause = [cond, *params]
+      condition = "(" + conditions.reduce { |a, c| a + ") and (" + c } + ")"
```

Comment:
A matter of preference, but I prefer meaningful argument names to a, b, c as well as string interpolation to suming up strings.  For example:

```ruby
.reduce { |cond, res| "(#{cond}) and #{res}" }
```

Reply:
I think giving variables a longer names would be counter-productive here (like naming iterator variable in C for loop `iterator`), since their whole scope is a single line. Besides, using `a` for `accumulator` seems to be a standard practice in such short reduce statements.

As for the implementation of reduce operation, I can use interpolation. I have no preference whatsoever.